packages/frontend/src/redux/reducers/sign/index.js

```diff
@@ -62,10 +62,11 @@ const sign = handleActions({
     [multiplyGas]: (state, { payload: { transactions: transactionsString } }) => {
         const transactions = deserializeTransactionsFromString(transactionsString);
```

Comment:
1. If we're going to use values from the `transactions` property of this slice's redux state, we should use _only_ values from that property of this slice of the the redux state rather than mix+match the data from re-parsing `account.url` _and_ existing `sign.transactions` state; this allows us to be absolutely sure that we're not doing something incorrect by assuming that the data in every index of `sign.transactions` and each transaction's `actions` property are 100% identical between the parsed output of `account.url` and the current redux `sign.transactions` state;  and by doing so,`multiplyGas` doesn't need to take any arguments at all if we've made the decision to use the existing state since this handler has direct access to the existing state since it is a reducer on the `sign` slice.

2. Also, since you're still passing `transactions` that is computed from the `account.url` state (which is not updated by `multiplyGas`, right?) to the `calculateGasLimit()` call, I think that the `gasLimit` property in state is not being correctly updated with the current code -- this type of bug is another danger of using two different sources of truth in the same function to generate new state.

I think for this to be correct we'll need something like:
```javascript
const transactionsWithDoubledGas = _.cloneDeep(state.transactions)
    .forEach((transaction) => {
        transactions.actions && transaction.actions.forEach((action) => {
            if(a.functionCall && a.functionCall.gas) {
                a.functionCall.gas = a.functionCall.gas.mul(new BN(MULTIPLY_TX_GAS_BY));
            }
        })
    });


return {
    ...state,
    transactions: transactionsWithDoubledGas,
    fees: {
        ...state.transactions.fees,
        gasLimit: calculateGasLimit(transactionsWithDoubledGas.flatMap(t => t.actions))
    }
}
```

This is assuming that `cloneDeep()` properly clones transaction objects _and_ any `bn.js` properties.  Otherwise you might need to explicitly call `.clone()` on any property of either Transaction and Action objects in the state if they are a BN.js class instance.

Reply:
Agree, Done :)